### ISSUE TITLE
Unify input handling between AptaTrans and AptaNet

### ISSUE THREAD

**Issue by ayushk687** [gc-os-ai/pyaptamer#664]
Summary

AptaTrans and AptaNet currently use separate input preprocessing and validation paths. This creates duplicated logic, inconsistent behavior, and unnecessary maintenance overhead.

This issue proposes introducing a unified input handling pipeline shared across both models.

Motivation

Current differences between the two pipelines may lead to:

inconsistent preprocessing behavior
duplicated tokenization logic
mismatched validation rules
harder integration for downstream users
increased maintenance cost

A shared interface would make both models easier to use and extend.

Proposed Changes
Create Shared Input Abstraction

Introduce a reusable input processing module/class that handles:

sequence validation
tokenization
padding/truncation
batching
tensor conversion
device placement
optional metadata handling

Example:
`class APIInputProcessor:
    def preprocess(self, aptamer, protein):
        ...`

Refactor Both Models

Update:

AptaTrans
AptaNet

To consume the same standardized input structure.

Potential approaches:

shared dataclass
common preprocessing utilities
unified dataset schema
reusable collate function

Expected Benefits
Reduced code duplication
Consistent inference behavior
Easier benchmarking across models
Cleaner public API
Simplified future model integrations
Improved maintainability

Suggested Scope
Included
preprocessing unification
shared validation
common batching behavior
standardized tensor formatting
Not Included
model architecture changes
retraining models
modifying prediction heads

**Comment by deveshsy** [comment]
I'd love to take ownership of this issue and implement the unified input handling pipeline. 

During the previous dataclass refactor discussions, I mapped out several of the critical bottlenecks between the `AptaTrans` and `AptaNet` entry points—specifically around data type coercion (handling `pd.DataFrame`, `list[tuple]`, and raw NumPy arrays cleanly) and ensuring functional purity so user data isn't mutated in-place during column mapping.

### Proposed Implementation Plan:
1. **Robust Coercion Layer:** Leverage a clean, centralized utility to canonicalize inputs into standard target formats before they hit the models, handling parameter checks gracefully.
2. **Defensive Validation Guardrails:** Build in fast-failing checks to prevent cryptic internal KeyDocs/KeyErrors (e.g., catching missing columns or header naming collisions gracefully if users pass dataframes with customized labels like `apta_col` or `prot_col`).
3. **Dynamic Metadata Tagging:** Ensure the underlying dataset container dynamically syncs its internal structural tags (like `has_y`) based on whether it's executing an unlabeled inference pipeline or a training routine, avoiding downstream PyTorch `DataLoader` batching crashes.

I can spin up a clean, tightly scoped PR targeting `main` that implements this shared abstraction without touching model architectures. Let me know if I can be assigned to this!
